### PR TITLE
UseTransformer - simplify/optimize

### PR DESCRIPTION
**Pull request by SpacePossum**
found when looking into;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4086

(does not fix it though)

### CODE REVIEW

File: src/Tokenizer/Transformer/UseTransformer.php [hunk 1]
```diff
@@ -51,22 +51,17 @@ public function process(Tokens $tokens, Token $token, $index)
     {
         if ($token->isGivenKind(T_USE) && $this->isUseForLambda($tokens, $index)) {
             $tokens[$index] = new Token([CT::T_USE_LAMBDA, $token->getContent()]);
-        }
 
-        if (!$token->isClassy()) {
```
Comment (SpacePossum) [inline src/Tokenizer/Transformer/UseTransformer.php]: narrowed down to `class` or `trait` as `interfaces` cannot import `traits`

File: src/Tokenizer/Transformer/UseTransformer.php [hunk 2]
```diff
@@ -51,22 +51,17 @@ public function process(Tokens $tokens, Token $token, $index)
...
             return;
         }
 
-        $prevTokenIndex = $tokens->getPrevMeaningfulToken($index);
-        $prevToken = null === $prevTokenIndex ? null : $tokens[$prevTokenIndex];
+        // Only search inside class/trait body for `T_USE` for traits.
+        // Cannot import traits inside interfaces or anywhere else
 
-        if ($prevToken->isGivenKind(T_DOUBLE_COLON)) {
```
Comment (SpacePossum) [inline src/Tokenizer/Transformer/UseTransformer.php]: this was done to check for `Foo::class`, however in such statement `class` is not a `T_CLASS` and so not `classy`

File: tests/Tokenizer/Transformer/UseTransformerTest.php [hunk 3]
```diff
@@ -75,6 +75,62 @@ public function provideProcessCases()
...
+                CT::T_USE_LAMBDA,
+                CT::T_USE_TRAIT,
+            ]
+        );
+    }
+
+    public function provideFix70Cases()
+    {
+        return [
+            'nested anonymous classes' => [
```
Comment (SpacePossum) [inline tests/Tokenizer/Transformer/UseTransformerTest.php]: this use case of nested anonymous classes didn't exists

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**